src/frontend/src/scheduler/distributed/stage.rs

```diff
@@ -364,6 +365,11 @@ impl StageRunner {
                         if let Some(stauts_res_inner) = status_res {
                             // The status can be Running, Finished, Failed etc. This stream contains status from
                             // different tasks.
+                            //
+                            //
+                            // Note: For Task execution failure, it now becomes a Rpc Error and will return here.
```

Comment:
Is this true?

Reply:
Yes.
```rust
let status = stauts_res_inner.map_err(|e| RpcError(e.into()))?;
```

Now task failure is treated as an Err, so it we turn into a RpcError and return immediately. I think I can add another error variant to make it better.

Reply:
Fixed